server/datastore/mysql/apple_mdm.go

```diff
@@ -385,20 +385,40 @@ func (ds *Datastore) IngestMDMAppleDevicesFromDEPSync(ctx context.Context, devic
 		return 0, ctxerr.Wrap(ctx, err, "ingest mdm apple host get app config")
 	}
 
+	args := []interface{}{nil}
+	if name := appCfg.MDM.AppleBMDefaultTeam; name != "" {
+		team, err := ds.TeamByName(ctx, name)
```

Comment:
What happens if the team name changes after the app config setting? Do we flow team name updates throughout the datastore? It seems like it would be easy for this to get lost inside the app config json.

Reply:
this is a good point, I will figure out what happens with the other configs that also require a team name (I _think_ we have others?)

but seems something that should be enforced (if we want to) when the config is set, not at this stage

Reply:
> but seems something that should be enforced (if we want to) when the config is set, not at this stage

Agreed. And I think we ought to have something to enforce this at the app config level. Is it possible for MySQL to have something like foreign key constraints inside json? If not, maybe we'd need something like we have for `hostRefs`?

As a precaution, I wonder if it would perhaps be a good idea as a general design policy that to save primary keys like team id inside json blobs in the datastore rather than values like name which might change.

Reply:
THis is something I mention here https://github.com/fleetdm/fleet/issues/8733#issuecomment-1325353478, the team might exist when the config is set, but can be deleted/renamed afterwards. In fact, nothing prevents a different team from being created later on with that name, and it would then be used for enrollment.

> seems something that should be enforced (if we want to) when the config is set

It is validated when the config is set IIRC, but the problem is that the team can be renamed/deleted soon after.